lesson 6/Tensor/app/src/main/cpp/native-lib.cpp

```diff
@@ -0,0 +1,95 @@
+#include <jni.h>
+#include <string>
+#include <string.h>
+#include <sys/socket.h>	//socket
+#include <arpa/inet.h>	//inet_addr
+#include <unistd.h>
+#include <android/log.h>
+#include <errno.h>
+#include <locale>
+
+
+#define LOG_TAG "CLIENT_APP"
+
+jbyteArray getImage(jobject context, JNIEnv* env) {
+    jclass contextThemeWrapperClass = env->FindClass("android/view/ContextThemeWrapper");
+    jclass assetManagerClass = env->FindClass("android/content/res/AssetManager");
+    jclass inputStreamClass = env->FindClass("java/io/InputStream");
+
+    jmethodID getAssetsID = env->GetMethodID(contextThemeWrapperClass, "getAssets", "()Landroid/content/res/AssetManager;");
+    jmethodID openID = env->GetMethodID(assetManagerClass, "open", "(Ljava/lang/String;)Ljava/io/InputStream;");
```

Comment:
Your implementation of this line was like this:
`jmethodID openFile = env->GetMethodID(AssetManager, "open", "([type)Ljava/io/InputStream;");`
There are a few issues here:
1. `jmethodID`, `jfieldID`  should be named by the function name (field name) plus 'ID', so it should be `openID` here.
2. It takes a String argument so the signature is "Ljava/lang/String;"